docs/plugins.md

```diff
@@ -0,0 +1,352 @@
+# Plugin Handbook
+
+## Table of Contents
+
+1. [Introduction](#introduction)
+   1. [What is a Lighthouse Plugin?](#what-is-a-lighthouse-plugin)
+   1. [Plugins v. Custom Config](#plugins-v-custom-config)
+   1. [Getting Started](#getting-started)
+1. [API](#api)
+   1. [Plugin Config](#plugin-config)
+   1. [Plugin Audits](#plugin-audits)
+1. [Best Practices](#best-practices)
+   1. [Naming](#naming)
+   1. [Scoring](#scoring)
+   1. [Common Mistakes](#common-mistakes)
+1. [Examples](#examples)
+
+## Introduction
+
+If you're new to Lighthouse development, start by reading up on the overall [architecture](./architecture.md) and how [configuration](./configuration.md) works before continuing.
+
+### What is a Lighthouse Plugin?
+
+Lighthouse plugins are a way to extend the functionality of Lighthouse with insight from domain experts (that's you!) and easily share this extra functionality with other Lighthouse users. At its core, a plugin is a node module that implements a set of checks that will get run by Lighthouse and added to the report as a new category.
+
+### Plugins v. Custom Config
+
+Plugins are easily shared and have a stable API that won't change between minor version bumps but are also more limited in scope. You might need a [custom Lighthouse configuration](./configuration.md) instead. Before getting started with plugins, think about your current needs, and consult the table below to decide which is best for you.
+
+| Capability                           | Plugin | Custom Config |
+| ------------------------------------ | ------ | ------------- |
+| Add custom audits                    | ✅      | ✅            |
+| Add custom categories                | ✅      | ✅            |
+| Easily sharable via NPM              | ✅      | ❌            |
+| Semver-stable API                    | ✅      | ❌            |
+| Gather custom data from audited page | ❌      | ✅            |
+| Modify core categories               | ❌      | ✅            |
+| Modify `config.settings` properties  | ❌      | ✅            |
+
+### Getting Started
+
+To develop a Lighthouse plugin, you'll need to write three things:
+
+1. A `package.json` file to define your plugin's dependencies and point to.
```

Comment:
and point to...what? :) entry point? `plugin.js`?

Reply:
heh yeah :)